packages/edit-site/src/store/actions.js

```diff
@@ -509,3 +514,26 @@ export const switchEditorMode =
 			speak( __( 'Mosaic view selected' ), 'assertive' );
 		}
 	};
+
+/**
+ * Action that switches the canvas mode.
+ *
+ * @param {?string} mode Canvas mode.
+ */
+export const __unstableSetCanvasMode =
+	( mode ) =>
+	( { registry, dispatch } ) => {
+		dispatch( {
+			type: 'SET_CANVAS_MODE',
+			mode,
+		} );
+		// Check if the block list view should be open by default.
+		if (
+			mode === 'edit' &&
+			registry
+				.select( preferencesStore )
+				.get( 'core/edit-site', 'showListViewByDefault' )
+		) {
+			dispatch.setIsListViewOpened( true );
```

Comment:
I think if we are on mobile, we should not do this. It feels strange to immediately open the list view on mobile as the list view will appear right away when the edit is pressed.

Reply:
This is the same code that was before on edit-site/src/index.js I think I prefer to leave it as is for now as it's kind of unrelated with the PR.